### ISSUE TITLE
Make media upload limits configurable

### ISSUE THREAD

**Issue by HorayNarea** [mastodon/mastodon#20490]
### Pitch

The variables in https://github.com/mastodon/mastodon/blob/9bc0a6c861e07d0112ef2e5ccd28adeca868bdbe/app/models/media_attachment.rb#L39-L45 should be overwritable from the configuration, so instance admins can decide for themself what limits they set.

### Motivation

This is needed because different instances/users have different requirements.

Especially `MAX_VIDEO_MATRIX_LIMIT` currently does not even allow direct upload of screen recordings of some phones.

**Comment by hsn10** [comment]
Make also configurable maximum attachments per user across all posts (if not already done).

**Comment by l1quid8** [comment]
I agree. Not every admin is concerned with having their instance federated, so the propagation error claim shouldn't be an excuse.
